src/checkbox.test.jsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import TestUtils from 'react-addons-test-utils';
+
+import Checkbox from './Checkbox';
+
+
+describe('Checkbox', function() {
+
+	let checkbox,
+		checkboxNode;
+
+	beforeEach(() => {
+		checkbox = TestUtils.renderIntoDocument(
+			<Checkbox label='Hello!' name='greeting' id='hello' value='hello' />
+		);
+		checkboxNode = ReactDOM.findDOMNode(checkbox);
+	});
+
+	afterEach(() => {
+		checkbox = null;
+		checkboxNode = null;
+	});
+
+	it('exists', function() {
+		expect(checkboxNode).not.toBeNull();
+	});
+
+	it('has a label with correct for attribute', function() {
+		const labelEl = checkboxNode.querySelector('label');
+		expect(labelEl.getAttribute('for')).toEqual('hello');
+	});
+
+	it('has a correct for attribute with generated id if no id is given', function() {
+		const name = 'greeting',
+			value = 'hello';
+		checkbox = TestUtils.renderIntoDocument(
+			<Checkbox label='Hello!' name={name} value={value} checked={false} />
+		);
+		checkboxNode = ReactDOM.findDOMNode(checkbox);
+		const checkboxEl = checkboxNode.querySelector('input[type=checkbox]');
+		const labelEl = checkboxNode.querySelector('label');
+
+		expect(checkboxEl.getAttribute('id')).toEqual(`${name}-${value}`);
+		expect(labelEl.getAttribute('for')).toEqual(`${name}-${value}`);
+	});
+
+
+	it('should be checked when specified', function() {
+		checkbox = TestUtils.renderIntoDocument(<Checkbox name='greeting' id='hello' value='hello' checked />);
+		checkboxNode = ReactDOM.findDOMNode(checkbox);
```

Comment:
We have been going back and forth on a few tests using findDOMNode so not gonna mark this as a change request but... offering this as an alternative approach to writing these tests. You can also read more about this [http://reactkungfu.com/2015/07/approaches-to-testing-react-components-an-overview/](here) 

```
it("renders a list in a box with proper CSS classes", function() {
    let box = this.TestUtils.findRenderedDOMComponentWithTag(this.component, "div");
    expect(box.props.className).toEqual("invitation-list-container");

    let list = this.TestUtils.findRenderedDOMComponentWithTag(box, "ul");
    expect(list.props.className).toEqual("invitation-list");
  });
```